### ISSUE TITLE
May I ask the version of PyTorch you used?

### ISSUE THREAD

**Issue by itaowei** [THUDM/GLM-iprompt#2]
When I try to run as described in [this](https://github.com/THUDM/GLM-iprompt#run-assuming-using-gpu0), there are always some errors about `torch`.
I have tried some different versions of PyTorch but it still cannot run successfully.
May I ask the version of PyTorch you used? If you can give me a docker image to run these commands, I will appreciate it a lot.

**Comment by xz-keg** [comment]
A full list of ``pip list`` of my environment is listed below.

For torch, I guess it is torch 1.11.0a0+b6df043.  

Package                       Version
----------------------------- ---------------------
absl-py                       1.0.0
aiohttp                       3.8.1
aiosignal                     1.2.0
alabaster                     0.7.12
apex                          0.1
appdirs                       1.4.4
argon2-cffi                   21.1.0
asgiref                       3.4.1
astor                         0.8.1
async-timeout                 4.0.2
attrs                         21.2.0
audioread                     2.1.9
Babel                         2.9.1
backcall                      0.2.0
backports.functools-lru-cache 1.6.4
beautifulsoup4                4.10.0
bleach                        4.1.0
blessings                     1.7
blis                          0.7.5
boto3                         1.20.21
botocore                      1.23.21
brotlipy                      0.7.0
cachetools                    4.2.4
catalogue                     2.0.6
certifi                       2021.10.8
cffi                          1.15.0
chardet                       4.0.0
charset-normalizer            2.0.0
click                         8.0.3
cloudpickle                   2.0.0
codecov                       2.1.12
colorama                      0.4.4
conda                         4.10.3
conda-build                   3.21.6
conda-package-handling        1.7.3
coverage                      6.1.2
cryptography                  35.0.0
cudf                          0+untagged.1.g0a00579
cugraph                       0+untagged.1.gfa6f0f1
cuml                          0+untagged.1.g273bc3c
cupy-cuda114                  9.3.0
cycler                        0.11.0
cymem                         2.0.6
Cython                        0.29.24
dask                          2021.7.1
dask-cuda                     21.8.0
dask-cudf                     0+untagged.1.g0a00579
dask-glm                      0.2.0
dask-ml                       1.9.0
dataclasses                   0.8
datasets                      1.17.0
debugpy                       1.5.1
decorator                     5.1.0
deepspeed                     0.5.8
defusedxml                    0.7.1
dill                          0.3.4
distributed                   2021.7.1
Django                        3.2.6
docutils                      0.17.1
entrypoints                   0.3
expecttest                    0.1.3
fastavro                      1.4.7
fastrlock                     0.8
fasttext                      0.9.2
filelock                      3.3.2
flake8                        3.7.9
Flask                         2.0.2
frozenlist                    1.2.0
fsspec                        2021.11.0
future                        0.18.2
glob2                         0.7
google-auth                   2.3.3
google-auth-oauthlib          0.4.6
gpustat                       0.6.0
graphsurgeon                  0.4.5
graphviz                      0.19
grpcio                        1.41.1
gunicorn                      20.1.0
h11                           0.12.0
h5py                          3.6.0
HeapDict                      1.0.1
hjson                         3.0.2
httptools                     0.2.0
huggingface-hub               0.4.0
hypothesis                    4.50.8
idna                          3.1
imagesize                     1.3.0
importlib-resources           5.4.0
iniconfig                     1.1.1
ipdb                          0.13.9
ipykernel                     6.5.0
ipython                       7.29.0
ipython-genutils              0.2.0
ipywidgets                    7.6.5
itsdangerous                  2.0.1
jedi                          0.18.0
Jinja2                        3.0.3
jmespath                      0.10.0
joblib                        1.1.0
json5                         0.9.6
jsonlines                     3.0.0
jsonschema                    4.2.1
jupyter                       1.0.0
jupyter-client                7.0.6
jupyter-console               6.4.0
jupyter-core                  4.9.1
jupyter-tensorboard           0.2.0
jupyterlab                    2.3.2
jupyterlab-pygments           0.1.2
jupyterlab-server             1.2.0
jupyterlab-widgets            1.0.2
jupytext                      1.13.1
kiwisolver                    1.3.2
langcodes                     3.2.1
libarchive-c                  3.1
librosa                       0.8.1
llvmlite                      0.36.0
lmdb                          1.2.1
locket                        0.2.1
Markdown                      3.3.4
markdown-it-py                1.1.0
MarkupSafe                    2.0.1
matplotlib                    3.4.3
matplotlib-inline             0.1.3
mccabe                        0.6.1
mdit-py-plugins               0.2.8
mistune                       0.8.4
mock                          4.0.3
mpi4py                        3.1.3
msgpack                       1.0.2
multidict                     5.2.0
multipledispatch              0.6.0
multiprocess                  0.70.12.2
murmurhash                    1.0.6
nbclient                      0.5.6
nbconvert                     6.2.0
nbformat                      5.1.3
nest-asyncio                  1.5.1
networkx                      2.0
ninja                         1.10.2.3
nltk                          3.6.5
notebook                      6.4.1
numba                         0.53.1
numpy                         1.21.4
nvidia-dali-cuda110           1.7.0
nvidia-dlprof-pytorch-nvtx    1.7.0
nvidia-dlprofviewer           1.7.0
nvidia-ml-py3                 7.352.0
nvidia-pyindex                1.0.9
nvtx                          0.2.3
oauthlib                      3.1.1
onnx                          1.10.1
packaging                     21.0
pandas                        1.2.5
pandocfilters                 1.5.0
parso                         0.8.2
partd                         1.2.0
pathy                         0.6.1
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        8.2.0
pip                           21.3.1
pkginfo                       1.7.1
pluggy                        1.0.0
polygraphy                    0.33.0
pooch                         1.5.2
preshed                       3.0.6
prettytable                   2.4.0
prometheus-client             0.12.0
prompt-toolkit                3.0.22
protobuf                      3.19.1
psutil                        5.8.0
ptyprocess                    0.7.0
py                            1.11.0
py3nvml                       0.2.7
pyarrow                       4.0.1
pyasn1                        0.4.8
pyasn1-modules                0.2.8
pybind11                      2.8.1
pycocotools                   2.0+nv0.5.1
pycodestyle                   2.5.0
pycosat                       0.6.3
pycparser                     2.20
pydantic                      1.8.2
pydot                         1.4.2
pyflakes                      2.1.1
Pygments                      2.10.0
pynvml                        11.0.0
pyOpenSSL                     21.0.0
pyparsing                     3.0.5
pyrsistent                    0.18.0
PySocks                       1.7.1
pytest                        6.2.5
pytest-cov                    3.0.0
pytest-pythonpath             0.7.3
python-dateutil               2.8.2
python-dotenv                 0.19.2
python-hostlist               1.21
python-nvd3                   0.15.0
python-slugify                5.0.2
pytorch-quantization          2.1.2
pytz                          2021.3
PyYAML                        6.0
pyzmq                         22.3.0
qtconsole                     5.2.1
QtPy                          1.11.3
regex                         2021.11.10
requests                      2.26.0
requests-oauthlib             1.3.0
resampy                       0.2.2
revtok                        0.0.3
rmm                           0+untagged.1.gccd9647
rouge                         1.0.1
rouge-score                   0.0.4
rsa                           4.7.2
ruamel-yaml-conda             0.15.80
s3transfer                    0.5.0
sacremoses                    0.0.46
scikit-learn                  0.23.1
scipy                         1.6.3
Send2Trash                    1.8.0
sentencepiece                 0.1.96
setuptools                    59.5.0
shellingham                   1.4.0
six                           1.16.0
sklearn                       0.0
smart-open                    5.2.1
snowballstemmer               2.1.0
sortedcontainers              2.4.0
SoundFile                     0.10.3.post1
soupsieve                     2.3
spacy                         3.2.0
spacy-legacy                  3.0.8
spacy-loggers                 1.0.1
Sphinx                        4.3.0
sphinx-glpi-theme             0.3
sphinx-rtd-theme              1.0.0
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
sqlparse                      0.4.2
srsly                         2.4.2
SwissArmyTransformer          0.1.2
tabulate                      0.8.9
tblib                         1.7.0
tensorboard                   2.7.0
tensorboard-data-server       0.6.1
tensorboard-plugin-wit        1.8.0
tensorboardX                  1.8
tensorrt                      8.0.3.4
terminado                     0.12.1
testpath                      0.5.0
text-unidecode                1.3
thinc                         8.0.12
threadpoolctl                 3.0.0
tokenizers                    0.10.3
toml                          0.10.2
tomli                         1.2.2
toolz                         0.11.2
torch                         1.11.0a0+b6df043
torch-tensorrt                1.0.0a0
torchtext                     0.12.0a0
torchvision                   0.11.0a0
tornado                       6.1
tqdm                          4.62.3
traitlets                     5.1.1
transformers                  4.15.0
treelite                      1.3.0
treelite-runtime              1.3.0
triton                        1.1.1
typer                         0.4.0
typing-extensions             3.10.0.2
ucx-py                        0+untagged.1.gbfa0450
uff                           0.6.9
Unidecode                     1.3.2
urllib3                       1.26.7
uvicorn                       0.15.0
uvloop                        0.16.0
wasabi                        0.8.2
watchgod                      0.7
wcwidth                       0.2.5
webencodings                  0.5.1
websockets                    10.0
Werkzeug                      2.0.2
wget                          3.2
wheel                         0.37.0
whitenoise                    5.3.0
widgetsnbextension            3.5.2
xgboost                       1.6.0.dev0
xmltodict                     0.12.0
xxhash                        2.0.2
yappi                         1.3.3
yarl                          1.7.2
zict                          2.0.0
zipp                          3.6.0

**Comment by itaowei** [comment]
Thanks a lot for your kind and quick reply!
